src/frontend/src/reducers/networks/utils/createTransitiveRoutes.ts

```diff
@@ -111,26 +111,27 @@ function addDataToPaths(path: PopulatedPath[]) {
             []
         );
 
-            segments.push({
-                fromStop: {
-                    coordinates: [boardStop.stop_lon, boardStop.stop_lat],
-                    name: boardStop.stop_name,
-                    stopId: boardStop.stop_id,
-                },
-                coordinates: latLons.map(([lat, lon]) => [lon, lat]), // reverse coords
-            mode: leg.route.route_type
-                ? TYPE_TO_ICON[leg.route.route_type]
-                : "",
-                name: toUpperCase(leg.route.route_short_name),
-                patternId: leg.pattern.pattern_id,
+        segments.push({
+            fromStop: {
+                coordinates: [boardStop.stop_lon, boardStop.stop_lat],
+                name: boardStop.stop_name,
+                stopId: boardStop.stop_id,
+            },
+            coordinates: latLons.map(([lat, lon]) => [lon, lat]), // reverse coords
+            mode:
+                leg.route.route_type || leg.route.route_type === 0
```

Comment:
There's some formatting that was captured in this commit, but the only edit is updating the conditional check for `mode` from `leg.route.route_type` to `leg.route.route_type || leg.route.route_type === 0`. This is supposed to be an index value, but the `0` value was incorrectly evaluated as false-y.